pilot/cmd/pilot-discovery/server/server.go

```diff
@@ -0,0 +1,391 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/davecgh/go-spew/spew"
+	"github.com/golang/glog"
+	durpb "github.com/golang/protobuf/ptypes/duration"
+	"github.com/hashicorp/go-multierror"
+	meshconfig "istio.io/api/mesh/v1alpha1"
+	configaggregate "istio.io/istio/pilot/adapter/config/aggregate"
+	"istio.io/istio/pilot/adapter/config/crd"
+	"istio.io/istio/pilot/adapter/config/file"
+	"istio.io/istio/pilot/adapter/config/ingress"
+	"istio.io/istio/pilot/adapter/config/memory"
+	"istio.io/istio/pilot/adapter/serviceregistry/aggregate"
+	"istio.io/istio/pilot/cmd"
+	"istio.io/istio/pilot/model"
+	"istio.io/istio/pilot/platform"
+	"istio.io/istio/pilot/platform/consul"
+	"istio.io/istio/pilot/platform/eureka"
+	"istio.io/istio/pilot/platform/kube"
+	"istio.io/istio/pilot/proxy"
+	"istio.io/istio/pilot/proxy/envoy"
+	"istio.io/istio/pilot/test/mock"
+	"istio.io/istio/pilot/tools/version"
+)
+
+// ServiceRegistry is an expansion of the platform.ServiceRegistry enum that adds a mock registry.
+type ServiceRegistry string
+
+const (
+	// MockRegistry environment flag
+	MockRegistry ServiceRegistry = "Mock"
+	// KubernetesRegistry environment flag
```

Comment:
May be Call this as inmemory registry?

Reply:
This registry hard-codes the hello and world services.  Simply `in-memory` doesn't quite capture that.

Reply:
If you're interested, you could create a memory registry, but it needs to be populated somehow.

Reply:
right that was my thought as well .... I just wanted to limit the scope of the change in this PR.  Would be an easy follow-on effort though.